src/app/components/product/productsList/ProductsListContainer.tsx

```diff
@@ -1,9 +1,17 @@
 import { ProductsList } from './ProductsList';
-import { useFetch } from 'hooks';
+import { useFetch, useUrl } from 'hooks';
 import { Product } from 'api/product/product.types';
+import { ProductsListContianerProps } from './ProductsListContainer.types';
+import { useEffect } from 'react';
 
-export const ProductsListContainer = () => {
-  const { data, isLoading, isError } = useFetch<Product[]>(process.env.REACT_APP_API_URL);
+export const ProductsListContainer = ({ setProductsCount }: ProductsListContianerProps) => {
+  const { queryParams } = useUrl();
+
+  const { data, isLoading, isError, meta } = useFetch<Product[]>(process.env.REACT_APP_API_URL, queryParams);
```

Comment:
Zamiast wrzucać `queryParams` tutaj, lepiej mieć funkcje, która wygeneruje konkretnego urla. W takim przypadku w `src/api/product/product.ts` stworzyć
```
const createGetProductsUrl = (params: GetProductsUrlParams): string => ...
```
gdzie `GetProductsUrlParams` będzie miał te parametry paginacji.
To jest o tyle korzystne, że w przypadku bardziej skomplikowanych zapytań (lub updatów istniejących danych w API) mamy dodatkową warstwę abstrakcji, w której możemy modyfikować dane z apki do istniejącego kontraktu z REST API.